packages/common/util/src/types.ts

```diff
@@ -30,18 +29,28 @@ export const isNotNullOrUndefined = <T>(value: T): value is Exclude<T, null | un
 // export const isNotNullish = <T>(value: T | null | undefined): value is T => value !== undefined && value !== null;
 export const boolGuard = <T>(value: T | null | undefined): value is T => Boolean(value);
 
+// TODO(burdon): Replace use of setTimeout everywhere?
+//  Would remove the need to cancel (and associated errors), but would change the operation of the code
+//  since the function would call immediately instead of waiting for the next tick.
+//  Could this affect performance? Otherwise replace with queueMicrotask?
+export const doAsync = async (fn: () => Promise<void>) => fn();
```

Comment:
I have reservations about this.

Timing matters in React and Code-Mirror:
   - React and other frameworks often rely on updates occurring in the next tick to maintain consistency.
   - Immediate execution could lead to unexpected behavior or even break component rendering in some cases.

Performance considerations:
   - Immediate execution might improve performance in some cases but could potentially block the main thread.

Instead of a blanket replacement, I suggest we continue with the status quo:
   - Evaluate each use case individually.
   - Use `setTimeout` for tasks that may need cancellation or deliberate delay.
   - Consider `queueMicrotask` for high-priority tasks that must run before the next render.
   - Retain immediate execution only where it's certain to be safe and beneficial and use `void` to indicate this in the code.

Reply:
thanks. yes definitely not planning to replace existing code, but i am questioning my own reflex to call setTimeout whenever i need to call an async function.